### ISSUE TITLE
Add support for setting / getting Cookies with SameSite attribute

### ISSUE THREAD

**Issue by nveeser** [SeleniumHQ/htmlunit-driver#97]
Update the driver to support testing with the SameSite attribute

**Comment by rbri** [comment]
Can  you please describe the expectations in more details.

**Comment by nveeser** [comment]
(Sorry should have given a little more there)

Modern browsers support cookies having an extra SameSite attribute.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite
RFC: https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-05

When using the current htmlunit-driver with an existing WebDriver implementation, if the cookie sent contains that attribute it is ignored / dropped. When reading back the cookie that attribute is not present.

**Comment by rbri** [comment]
Thanks for the report, will try to add this to the next HtmlUnit Version

**Comment by rbri** [comment]
have done a first commit (https://github.com/HtmlUnit/htmlunit/commit/7ec4b17d97373f8acc59cf85d2055f1c0ee81483) if you like you can help by providing more detailed test cases.

**Comment by rbri** [comment]
@nveeser have done some impl and written some tests. It will be great, if you can have a look at the test and give me some hints what is missing.

**Comment by nveeser** [comment]
Sure... so my use case is for unit testing the go-lang webriver implementation here
https://github.com/tebeka/selenium

Can you point me to some basic docs for building the candidate jar? Then I can use that to re-run my tests and see if they now pass.

**Comment by rbri** [comment]
There is no need to build it yourself - you can simply download from the CI server
https://jenkins.wetator.org/view/HtmlUnit/job/HtmlUnit%20-%20Headless/ws/target/

**Comment by nveeser** [comment]
So I downloaded and tried the unit tests again.
(I am a bit of a novice with WebDriver in general so bear with me.)

When I run things, the selenium remote cannot initialize a session
---
The golang unit tests start the server using:
java -cp vendor/htmlunit-driver.jar:vendor/selenium-server.jar
org.openqa.grid.selenium.GridLauncherV3 -port 42657 -debug

It then calls:
POST /wd/hub/session
{"capabilities":{"alwaysMatch":{"browserName":"htmlunit"}},"desiredCapabilities":{"browserName":"htmlunit","javascriptEnabled":true}}

Gets an exception from selenium remote (in JSON format)

    "stackTrace": [
      {
        "fileName": "NewSessionPipeline.java",
        "methodName": "lambda$null$4",
        "className": "org.openqa.selenium.remote.server.NewSessionPipeline",
        "lineNumber": 76
      },
      {
        "fileName": "Optional.java",
        "methodName": "orElseThrow",
        "className": "java.util.Optional",
        "lineNumber": 409
      },
      {
        "fileName": "NewSessionPipeline.java",
        "methodName": "lambda$createNewSession$5",
        "className": "org.openqa.selenium.remote.server.NewSessionPipeline",
        "lineNumber": 75
      },
      {
        "fileName": "Optional.java",
        "methodName": "orElseGet",
        "className": "java.util.Optional",
        "lineNumber": 370
      },
      {
        "fileName": "NewSessionPipeline.java",
        "methodName": "createNewSession",
        "className": "org.openqa.selenium.remote.server.NewSessionPipeline",
        "lineNumber": 73
      },
      {
        "fileName": "BeginSession.java",
        "methodName": "execute",
        "className":
"org.openqa.selenium.remote.server.commandhandler.BeginSession",
        "lineNumber": 65
      },
      {
        "fileName": "WebDriverServlet.java",
        "methodName": "lambda$handle$0",
        "className": "org.openqa.selenium.remote.server.WebDriverServlet",
        "lineNumber": 235
      },
      {
        "fileName": "Executors.java",
        "methodName": "call",
        "className": "java.util.concurrent.Executors$RunnableAdapter",
        "lineNumber": 515
      },
      {
        "fileName": "FutureTask.java",
        "methodName": "run",
        "className": "java.util.concurrent.FutureTask",
        "lineNumber": 264
      },
      {
        "fileName": "ThreadPoolExecutor.java",
        "methodName": "runWorker",
        "className": "java.util.concurrent.ThreadPoolExecutor",
        "lineNumber": 1128
      },
      {
        "fileName": "ThreadPoolExecutor.java",
        "methodName": "run",
        "className": "java.util.concurrent.ThreadPoolExecutor$Worker",
        "lineNumber": 628
      },
      {
        "fileName": "Thread.java",
        "methodName": "run",
        "className": "java.lang.Thread",
        "lineNumber": 835
      }
    ],


Any suggestions what I am doing wrong here?

**Comment by nveeser** [comment]
For clarity here is the Java stacktrace directly

java.version: '11.0.10'\nDriver info: driver.version: unknown
at
org.openqa.selenium.remote.server.NewSessionPipeline.lambda$null$4(NewSessionPipeline.java:76)
at java.base\u002fjava.util.Optional.orElseThrow(Optional.java:409)
at
org.openqa.selenium.remote.server.NewSessionPipeline.lambda$createNewSession$5(NewSessionPipeline.java:75)
at java.base\u002fjava.util.Optional.orElseGet(Optional.java:370)
at
org.openqa.selenium.remote.server.NewSessionPipeline.createNewSession(NewSessionPipeline.java:73)
at
org.openqa.selenium.remote.server.commandhandler.BeginSession.execute(BeginSession.java:65)
at
org.openqa.selenium.remote.server.WebDriverServlet.lambda$handle$0(WebDriverServlet.java:235)
at
java.base\u002fjava.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
at java.base\u002fjava.util.concurrent.FutureTask.run(FutureTask.java:264)
at
java.base\u002fjava.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at
java.base\u002fjava.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base\u002fjava.lang.Thread.run(Thread.java:835)\n",

**Comment by rbri** [comment]
From the first look - i have no idea.

In between i made a new release.

**Comment by rbri** [comment]
@nveeser  Any news here?

**Comment by jvanhill** [comment]
@rbri - Didn't see that a fix for this issue was ever merged. I am seeing where same-site cookie is being dropped.

See it both in 2.44 and 2.59.

**Comment by rbri** [comment]
@jvanhill how do you read the cookie

**Comment by jvanhill** [comment]
@rbri - Thanks for the response. I resolved this issue late last week. I had been running with an older version of htmlunit and moved up to 2.59 when I found this defect. However; there was another build task that was replacing my htmlunit library with the old version. Once I took care of that, I could now see the cookie with samesite.

I can verify that the fix worked if you want to close this issue.
